packages/main/src/TextArea.ts

```diff
@@ -0,0 +1,707 @@
+	 *
+	 * <br><br>
+	 * <b>Note:</b> If not specified, a default text (in the respective language) will be displayed.
+	 *
+	 * <br><br>
+	 * <b>Note:</b> The <code>valueStateMessage</code> would be displayed if the component has
+	 * <code>valueState</code> of type <code>Information</code>, <code>Warning</code> or <code>Error</code>.
+	 * @type {HTMLElement[]}
+	 * @name sa.ui.webc.main.TextArea.prototype.valueStateMessage
+	 * @since 1.0.0-rc.7
+	 * @slot
+	 * @public
+	 */
+	@slot()
+	valueStateMessage!: Array<HTMLElement>;
+	/**
+	 * The slot is used to render native <code>input</code> HTML element within Light DOM to enable form submit,
+	 * when <code>name</code> property is set.
+	 * @type {HTMLElement[]}
+	 * @name sa.ui.webc.main.TextArea.prototype.formSupport
+	 * @slot
+	 * @private
+	 */
+	 @slot()
+	 formSupport!: Array<HTMLElement>;
+
+	_fnOnResize: ResizeObserverCallback;
+	_firstRendering: boolean;
+	_openValueStateMsgPopover: boolean;
+	_exceededTextProps!: ExceededText;
+	_keyDown?: boolean;
+	FormSupport?: typeof FormSupportT;
+	previousValue: string;
+	popover?:Popover;
+
+	static get styles() {
+		return [browserScrollbarCSS, styles];
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get template() {
+		return TextAreaTemplate;
+	}
+
+	static get staticAreaTemplate() {
+		return TextAreaPopoverTemplate;
+	}
+
+	static get staticAreaStyles() {
+		return valueStateMessageStyles;
+	}
+
+	static i18nBundle: I18nBundle;
+
+	static async onDefine() {
+		TextArea.i18nBundle = await getI18nBundle("@ui5/webcomponents");
+	}
+
+	constructor() {
+		super();
+
+		this._firstRendering = true;
+		this._openValueStateMsgPopover = false;
+		this._fnOnResize = this._onResize.bind(this);
+		this.previousValue = "";
+	}
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._fnOnResize);
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._fnOnResize);
+	}
+
+	onBeforeRendering() {
+		this._exceededTextProps = this._calcExceededText();
+		this._mirrorText = this._tokenizeText(this.value);
+
+		this.exceeding = !!this._exceededTextProps.leftCharactersCount && this._exceededTextProps.leftCharactersCount < 0;
+		this._setCSSParams();
+
+		const FormSupport = getFeature<typeof FormSupportT>("FormSupport");
+		if (FormSupport) {
+			FormSupport.syncNativeHiddenInput(this);
+		} else if (this.name) {
+			console.warn(`In order for the "name" property to have effect, you should also: import "@ui5/webcomponents/dist/features/InputElementsFormSupport.js";`); // eslint-disable-line
+		}
+	}
+
+	onAfterRendering() {
+		const nativeTextArea = this.getInputDomRef()!;
+
+		if (this.rows === 1) {
+			nativeTextArea.setAttribute("rows", "1");
+		} else {
+			nativeTextArea.removeAttribute("rows");
+		}
+
+		this.toggleValueStateMessage(this.openValueStateMsgPopover);
+		this._firstRendering = false;
+	}
+
+	getInputDomRef() {
+		return this.getDomRef()!.querySelector("textarea");
```

Comment:
it would be even better to have it like this:
```ts
return this.getDomRef()!.querySelector<HTMLTextAreaElement>("textarea")!;
```

First, this will be guaranteed to return a text area instance, and the code that uses this function will be able to access text area specific properties, and second, the ! at the end guarantees it, so you don't have to always use ! when you call the function from other functions.